doozer/doozerlib/backend/rebaser.py

```diff
@@ -806,6 +806,11 @@ def _add_build_repos(self, dfp: DockerfileParser, metadata: ImageMetadata, dest_
         # 2. "cachito-emulation" - inject cachito-like environment variables & resources into Dockerfile.
         # 3. "cachito-removal" - Comment out lines referencing REMOTE_SOURCES.
         cachito_mode = metadata.config.konflux.cachito.mode
+        if metadata.prefetch_enabled():
+            # if prefetch is enabled, cachito lines (REMOTE_SOURCES etc.) should be commented out
+            self._logger.info("Setting cachito_mode = 'removal' since pre-fetch is enabled")
+            cachito_mode = 'removal'
```

Comment:
All modes were meant to be useful when cachi2 is enabled. Not sure why we are forcing removal.

Reply:
If cachi2 is enabled, then for konflux builds, we don't need cachito right?